### ISSUE TITLE
Compiler nullderef(??)

### ISSUE THREAD

**Issue by DutchGhost** [ziglang/zig#1421]
compiling the following code with build https://ci.appveyor.com/project/andrewrk/zig-d3l86/build/0.2.0+68e2794e, crashes the compiler.

```
const start = struct {
    fn crash() bug() { return bug; }
};

fn bug() void {
    _ = @typeInfo(start).Struct;
}

test "example" {
    var boom = start.crash();
}

pub fn main() void {
    var boom = start.crash();
}
```

**Comment by andrewrk** [comment]
Thanks for the small test case!

**Comment by DutchGhost** [comment]
An even smaller test case:

```
const bug = struct {
    fn crash() @import("builtin").TypeInfo {
        return @typeInfo(bug);
    }
};
test "example" {
    const boom = bug.crash();
}

pub fn main() void {
    const boom = bug.crash();
}
```

**Comment by andrewrk** [comment]
fixed in 009e90f446ec7a81d602906fda9b17a6af1859f5
